### PR TITLE
{rstantools}-based compilation

### PR DESCRIPTION
**Pull request by bearloga**
+ cleans up Stan model compilation by switching over to the [{rstantools}-based workflow ](https://mc-stan.org/rstantools/articles/minimal-rstan-package.html)for #1448
+ minor documentation change: {devtools} -> {remotes}, which is better for end-users
+ minor README formatting changes
+ adds RStudio project file which makes it easier for community to get started with contributing to the package

### CODE REVIEW

**Comment by bearloga** [comment]
Just signed the individual CLA

File: R/DESCRIPTION [hunk 1]
```diff
@@ -12,8 +12,10 @@ Description: Implements a procedure for forecasting time series data based on
     that have strong seasonal effects and several seasons of historical data.
     Prophet is robust to missing data and shifts in the trend, and typically
     handles outliers well.
+URL: https://github.com/facebook/prophet
+BugReports: https://github.com/facebook/prophet/issues
 Depends:
-    R (>= 3.2.3),
+    R (>= 3.4.0),
```
Comment (hadley) [inline R/DESCRIPTION:18]: Out of interest, why was this bump needed?

Comment (bearloga) [reply R/DESCRIPTION:18]: rstantools bumped it up. Seems rstan started requiring it in 2018 for C++14 https://github.com/stan-dev/rstan/commit/cd9257016132685858b01e7699d487b97cd25df8

Comment (hadley) [reply R/DESCRIPTION:18]: Interesting, thanks!

**Comment by seanjtaylor** [comment]
Thank you so much! Tested and works for me. Merging it now.

File: R/DESCRIPTION (hunk 1 above)
Comment (bearloga) [reply R/DESCRIPTION:18]: You’re welcome! :D

**Comment by bletham** [comment]
@bearloga this is really nice, a big thanks for the fix!

**Pull request merged**